src/Token/Generator.php

```diff
@@ -201,6 +217,10 @@ private function loadSigningKey(OpenSSLAsymmetricKey|string $signingKey, null|st
             throw TokenException::unableToProcessSigningKey(sprintf(TokenException::MSG_KEY_TYPE_NOT_SUPPORTED, $details['type'] ?? 'unknown'));
         }
 
+        if (! isset($details['bits']) || intval($details['bits']) < self::CONST_ALGO_MIN_LENGTHS[$this->algorithm] || intval($details['bits']) > self::CONST_ALGO_MAX_LENGTHS[$this->algorithm]) {
```

Comment:
Do you need validation for this on the client - I know that there are limits applied on the server when you upload the public credential

Reply:
Also the limit for the public key is 4kb (https://auth0team.atlassian.net/browse/IPS-2763) - So I think we support private keys that are larger than 4kb

Reply:
Removed length checks in https://github.com/auth0/auth0-PHP/pull/699/commits/5ed22572c6cf185b93f72b57931b339bc0985158